datafusion/sqllogictest/test_files/joins.slt

```diff
@@ -1336,6 +1350,25 @@ physical_plan
 10)----------RepartitionExec: partitioning=RoundRobinBatch(2), input_partitions=1
 11)------------MemoryExec: partitions=1, partition_sizes=[1]
 
+
+# Join on struct
+query ??
+select join_t3.s3, join_t4.s4
+from join_t3
+inner join join_t4 on join_t3.s3 = join_t4.s4
+----
+{id: 2} {id: 2}
```

Comment:
Maybe also add an `explain` query to make sure it is using HashJoin?

Reply:
Added for this query. The one using `IS NOT DISTINCT FROM` is still not using hash join. But that does not seem to be related to the struct, since it the same if just joining on a int.

Reply:
Hmm, so the query using `IS NOT DISTINCT FROM` is actually not testing HashJoin path.

Reply:
Sadly not. If we made some other improvements like https://github.com/apache/datafusion/pull/11272 it would.

But for not I just changed the test case to use `EXCEPT` instead and test that path. (The test case fails on master without the fixes in the branches)